From 5e7482f6b6edb7d35d6e0ba11d9240382add8489 Mon Sep 17 00:00:00 2001
From: David Daney <ddaney@caviumnetworks.com>
Date: Sun, 26 May 2019 13:16:08 +0200
Subject: [PATCH 158/345] MIPS: Octeon: Initialize proper CVMX_SSO_NW_TIM
 register.

CN68XX uses a different register address.  Make the register selection
depend on the chip type.

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Carlos Munoz <cmunoz@caviumnetworks.com>
---
 arch/mips/cavium-octeon/setup.c | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/arch/mips/cavium-octeon/setup.c b/arch/mips/cavium-octeon/setup.c
index f3352aa352bc..e2dbb6adc533 100644
--- a/arch/mips/cavium-octeon/setup.c
+++ b/arch/mips/cavium-octeon/setup.c
@@ -42,6 +42,7 @@
 #include <asm/octeon/octeon.h>
 #include <asm/octeon/pci-octeon.h>
 #include <asm/octeon/cvmx-rst-defs.h>
+#include <asm/octeon/cvmx-sso-defs.h>
 #include <asm/octeon/cvmx-qlm.h>
 #include <asm/octeon/cvmx-debug.h>
 
@@ -542,7 +543,8 @@ void octeon_user_io_init(void)
 			  CONFIG_CAVIUM_OCTEON_CVMSEG_SIZE,
 			  CONFIG_CAVIUM_OCTEON_CVMSEG_SIZE * 128);
 
-	if (octeon_has_feature(OCTEON_FEATURE_FAU)) {
+	if (current_cpu_type() != CPU_CAVIUM_OCTEON3 ||
+	    OCTEON_IS_MODEL(OCTEON_CN70XX)) {
 		union cvmx_iob_fau_timeout fau_timeout;
 
 		/* Set a default for the hardware timeouts */
@@ -553,9 +555,14 @@ void octeon_user_io_init(void)
 		cvmx_write_csr(CVMX_IOB_FAU_TIMEOUT, fau_timeout.u64);
 	}
 
-	if ((!OCTEON_IS_MODEL(OCTEON_CN68XX) &&
-	     !OCTEON_IS_MODEL(OCTEON_CN7XXX)) ||
-	    OCTEON_IS_MODEL(OCTEON_CN70XX)) {
+	if (OCTEON_IS_MODEL(OCTEON_CN68XX) || octeon_has_feature(OCTEON_FEATURE_CIU3)) {
+		union cvmx_sso_nw_tim nm_tim;
+
+		nm_tim.u64 = 0;
+		/* 4096 cycles */
+		nm_tim.s.nw_tim = 3;
+		cvmx_write_csr(CVMX_SSO_NW_TIM, nm_tim.u64);
+	} else {
 		union cvmx_pow_nw_tim nm_tim;
 
 		nm_tim.u64 = 0;
-- 
2.25.1

